### PR TITLE
Add support Hidden, JsonIgnore annotations for enum values

### LINKED ISSUE #825: Unable to hide an enum value in Swagger Docs
### Expected Behavior

The hidden annotated should not be displayed in swagger docs v3

Value1
Value3

### Actual Behaviour

The swagger docs shows all the values listed for that enum

Value1,
Value2,
Value3

### Steps To Reproduce

public enum Foo {
   Value1,
  @Hidden
   Value2,
    Value3
}

Annotation which I tried - @Hidden, @JsonIgnore, @Schema( allowableValues = "value1", "value3") 

None of the above annotation worked.

### Environment Information

- mac
- jdk 17

### Example Application

_No response_

### Version

3.1.3

### PR DESCRIPTION
**Pull request by altro3**
Fixes #825

### CODE REVIEW

**Comment by altro3** [comment]
@graemerocher Hi! Could you review and merge last 3 PR's? And, maybe, it's time to release?

**Comment by altro3** [comment]
@graemerocher Could you also merge this fix?

File: openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java [hunk 1]
```diff
@@ -2015,6 +2015,15 @@ protected Schema readSchema(AnnotationValue<io.swagger.v3.oas.annotations.media.
     private List<Object> getEnumValues(EnumElement type, String schemaType, String schemaFormat, VisitorContext context) {
         List<Object> enumValues = new ArrayList<>();
         for (EnumConstantElement element : type.elements()) {
+
+            AnnotationValue<io.swagger.v3.oas.annotations.media.Schema> schemaAnn = element.getAnnotation(io.swagger.v3.oas.annotations.media.Schema.class);
+            boolean isHidden = schemaAnn != null && schemaAnn.get("hidden", Boolean.class).orElse(false);
```
Comment (graemerocher) [inline openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java:2020]: can be written as `element.booleanValue(io.swagger.v3.oas.annotations.media.Schema.class, "hidden").orElse(false)`

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java (hunk 1 above)
Comment (altro3) [reply openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java]: @graemerocher fixed

**Comment by graemerocher** [comment]
Could you address this security hotspot https://sonarcloud.io/project/security_hotspots?id=micronaut-projects_micronaut-openapi&branch=master&resolved=false&types=SECURITY_HOTSPOT

**Comment by altro3** [comment]
@graemerocher fixed

**Pull request merged**